provider/provider.go

```diff
@@ -745,6 +745,15 @@ func (dc DependencyCondition) Evaluate(ctx context.Context, log logr.Logger, con
 			if err == nil {
 				incident.LineNumber = &location.StartPosition.Line
 				incident.CodeLocation = &location
+			} else if baseDep, ok := matchedDep.dep.Extras["baseDep"]; ok {
```

Comment:
since I looked into too many perf issues recently :)... maybe we can make only one call to GetLocation based on either direct or base dep?